source/code/plugin/in_kube_podinventory.rb

```diff
@@ -317,6 +317,19 @@ def parse_and_emit_records(podInventory, serviceList)
                   record["ContainerStatusReason"] = containerStatus[containerStatus.keys[0]]["reason"]
                 end
               end
+
+              # Record the last state of the container. This may have information on why a container was killed.
+              if container["lastState"].keys.length > 0
+                lastStateName = container["lastState"].keys[0]
+                record["ContainerLastStatus"] = Hash.new
```

Comment:
Again, you are assuming all these properties exist all the time. Is that true ? Can you please add check for existence ?